database/program_db.js

```diff
@@ -1,46 +1,83 @@
+const { validate } = require("../models/Program");
 const Program = require("../models/Program");
-const grades = ["6th", "7th", "8th", "9th", "10th", "11th", "12th", "Out of High School", "College Freshman", "College Sophmore", "College Junior", "College Senior", "Out of College"];
+const registration_db = require("./registration_db");
+const constants = require("../routes/constants")
+const grades = constants.getGradeLevels()
+const d = new Date();
 
 module.exports = {
 	addProgram: async function(programObj) {
-    if (grades.indexOf(programObj.min_grade_level) <= grades.indexOf(programObj.max_grade_level)) {
-      const newprogram = new Program({
-        title: programObj.title,
+    if (validateProgram(programObj)) {
+      registration = false;
+      if((typeof(programObj.registration_required)) == "undefined"){  
+        registration = false;
+      }
+      if(((typeof programObj.registration_required)) == "string" ){
+        registration = true;
+      }      
+      let year = d.getFullYear();
+      let month = d.getMonth();
+      title = (programObj.title + " (" + month + " - " + year + ")" )
+      const newProgram = new Program({
+        title: title,
         description: programObj.description,
         location: programObj.location,
         start_date: programObj.start_date,
         end_date: programObj.end_date,
         min_grade_level: programObj.min_grade_level,
         max_grade_level: programObj.max_grade_level,
-        program_id_number: `${programObj.title}.${ await module.exports.getTitleCount(programObj.title)}`,
+        registration_required: registration,
+        program_id: `${programObj.title}.${ await module.exports.getTitleCount(programObj.title)}`,
         status: "active"
-      });
-
-      await newprogram.save()
+      })
+      await newProgram.save()
+      let student_list = programObj.student_list
+      if(student_list !== undefined) {
+        if(student_list.length == 24) {
+          await registration_db.addRegistration(student_list, newProgram.id)
```

Comment:
when selecting one option for students involved, it turns the id into an array of chars, so when there is one item, 24 is used to check if it is one item and then adds the one item to the linked table (registration)

Reply:
hm, weird that the form control would send the data like that. in the places where this happens, i would recommend using

```js
student_list instanceof Array
```

or something like that instead, after checking if it's undefined. if `student_list` _is_ a list, you can add all the students from the list. otherwise you can just add the singular one.

Reply:
Otherwise if you happened to add 24 students to a program, it would get really weird